tools/azsdk-cli/Azure.Sdk.Tools.Cli/Helpers/PackageInfo/TypeScriptPackageInfoHelper.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.Text.RegularExpressions;
+
+namespace Azure.Sdk.Tools.Cli.Helpers;
+
+public sealed class TypeScriptPackageInfoHelper(IGitHelper gitHelper) : IPackageInfoHelper
+{
+    public Task<PackageInfo> ResolvePackageInfo(string packagePath, CancellationToken ct = default)
+    {
+        var realPath = RealPath.GetRealPath(packagePath);
+        var (repoRoot, relativePath, fullPath) = Parse(gitHelper, realPath);
+        var model = new PackageInfo
+        {
+            PackagePath = fullPath,
+            RepoRoot = repoRoot,
+            RelativePath = relativePath,
+            PackageName = Path.GetFileName(fullPath),
+            ServiceName = Path.GetFileName(Path.GetDirectoryName(fullPath)) ?? string.Empty,
+            Language = "typescript",
+            SamplesDirectoryProvider = (pi, _) => Task.FromResult(Path.Combine(pi.PackagePath, "samples-dev")),
+            FileExtensionProvider = _ => ".ts",
+            VersionProvider = (pi, token) => TryGetVersionAsync(pi.PackagePath, token)
+        };
+        return Task.FromResult(model);
+    }
+
+    private static async Task<string?> TryGetVersionAsync(string packagePath, CancellationToken ct)
+    {
+        try
+        {
+            var path = Path.Combine(packagePath, "package.json");
+            if (!File.Exists(path)) { return null; }
+            var content = await File.ReadAllTextAsync(path, ct);
+            var match = Regex.Match(content, "\"version\"\\s*:\\s*\"([^\\\"]+)\"");
```

Comment:
Could we parse the JSON instead? https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsondocument.parseasync?view=net-9.0